### PR TITLE
feat: add optional params to device.transport

### PR DESCRIPTION
**Pull request by Justkant**

### CODE REVIEW

**Review by chabroA — APPROVED** [review]

File: packages/core/src/spec/types/DeviceTransport.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { z } from "zod";
 
 const schemaDeviceTransportParams = z.object({
   appName: z.string().optional(),
+  appVersion: z.string().optional(),
```
Comment (chabroA) [inline packages/core/src/spec/types/DeviceTransport.ts:5]: Improvement, we could use a regex for the semver string
cf. the one recommended on [semver.org](https://semver.org) -> https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Comment (Justkant) [reply packages/core/src/spec/types/DeviceTransport.ts]: fixed

**Review by chabroA — APPROVED** [review]

File: packages/core/src/spec/types/DeviceTransport.ts [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+]);
+
+export type DeviceType = z.infer<typeof schemaDeviceType>;
+
 const schemaDeviceTransportParams = z.object({
   appName: z.string().optional(),
+  appVersionRange: z.string().optional(),
+  firmwareVersionRange: z.string().optional(),
+  seeded: z.boolean().optional(),
+  devices: z.array(schemaDeviceType).optional(),
```
Comment (chabroA) [inline packages/core/src/spec/types/DeviceTransport.ts:18]: Empty array should be forbidden as well

```suggestion
  devices: z.array(schemaDeviceType).nonempty().optional(),
```

**Review by chabroA — CHANGES_REQUESTED** [review]

File: packages/core/src/spec/types/DeviceTransport.ts (hunk 2 above)
Comment (Justkant) [reply packages/core/src/spec/types/DeviceTransport.ts:18]: done

**Review by chabroA — APPROVED** [review]

**Pull request merged**